source/string_tools.f90

```diff
@@ -256,6 +257,34 @@ subroutine chr_scanString(theString, theMask, maxLen, nValues, nIndent, hasErr)
 
 end subroutine chr_scanString
 
+! ================================================================================================ !
+!  Extract file name from path
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-02-19
+! ================================================================================================ !
+function chr_baseName(pathName) result(fileName)
+
+  character(len=*), intent(in)  :: pathName
+  character(len=mFileName) :: fileName
+  integer iC, iS, iE
+
+  iS = 1
+  iE = len_trim(pathName)
+  do iC=iE,1,-1
+    if(pathName(iC:iC) == "/" .or. pathName(iC:iC) == "\") then
+      iS = iC + 1
+      exit
+    end if
+  end do
+
+  if(iE-iS > mFileName) then
+    fileName = pathName(iS:iS+mFileName)
```

Comment:
Here you have a filename truncation - don't you want to raise an error message and stop processing?

Reply:
No, actually, I don't want to cause an error. The function is only used for the log file.

There is a point that someone else may want to use it to extract the file name though, in which case a truncated file name can be dangerous. Maybe we should return an error value at least so this can be parsed.

Reply:
that's exactly the case I was thinking about... there is no explicit warning in the function which limits its use to the log file (even if grepping through the code will pop up immediately the only concerned line), hence a user may want to re-use that function - since it is also located in a utility module :)